infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -325,14 +325,79 @@ private void applyShadowStream(UUID streamId, long snapshot) {
         }
     }
 
+    /**
+     * Get the shadow stream's name of given stream id, note that we are now using stream id instead
+     * of stream name as part of shadow stream's name.
+     *
+     * @param regularStreamId Given stream id
+     * @return Corresponding shadow stream's name
+     */
+    private String getShadowStreamName(UUID regularStreamId) {
+        return regularStreamId + SHADOW_STREAM_SUFFIX;
+    }
+
+    /**
+     * Get the shadow stream's UUID of an original stream from its UUID.
+     *
+     * @param regularStreamId UUID of the original stream
+     * @return UUID of the original stream's shadow stream
+     */
+    private UUID getShadowStreamUUID(UUID regularStreamId) {
```

Comment:
can we keep the name as getShadowStreamId()?  It is easier to remember and read..